server/src/main/java/org/opensearch/index/store/remote/filecache/FullFileCachedIndexInput.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.store.remote.filecache;
+
+import org.apache.lucene.store.AlreadyClosedException;
+import org.apache.lucene.store.IndexInput;
+import org.opensearch.common.annotation.ExperimentalApi;
+
+import java.nio.file.Path;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * Implementation of the CachedIndexInput for NON_BLOCK files which takes in an IndexInput as parameter
+ *
+ * @opensearch.experimental
+ */
+@ExperimentalApi
+public class FullFileCachedIndexInput implements CachedIndexInput {
+
+    private final IndexInput indexInput;
```

Comment:
we don't need the underlying `IndexInput` anymore, just keeping a reference to `FileCachedIndexInput` should be good enough.